src/core/midiMsg.cpp

```diff
@@ -0,0 +1,66 @@
+/* -----------------------------------------------------------------------------
+ *
+ * Giada - Your Hardcore Loopmachine
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2020 Giovanni A. Zuliani | Monocasual
+ *
+ * This file is part of Giada - Your Hardcore Loopmachine.
+ *
+ * Giada - Your Hardcore Loopmachine is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General
+ * Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * Giada - Your Hardcore Loopmachine is distributed in the hope that it
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Giada - Your Hardcore Loopmachine. If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------- */
+
+#include "midiMsg.h"
+#include <string>
+#include <vector>
+
+namespace giada {
+namespace m {
+
+//------------------------------  CONSTRUCTORS  --------------------------------
+
+MidiMsg::MidiMsg(std::string sender, std::vector<unsigned char>* message){ 
+	m_sender = sender;
+	m_message = message;
```

Comment:
My original reasoning was that MidiMsg is created and consumed before the context gets back to RtMidi, that originally created the message vector - so surely it won't be destroyed in the meantime.
On the other hand, I'm guessing this isn't much of a cost to copy these messages just to be safe, and hardened against the possible future uses. 
Good idea, I'll copy vector into MidiMsg rather than store a pointer.